lib/msf/core/post/vcenter/vcenter.rb

```diff
@@ -0,0 +1,702 @@
+            output.append(user)
+          end
+          output
+        end
+
+        #
+        # Returns a list of virtual machines located on the server
+        # @param pg_password [String] postgress password
+        # @param vcdb_user [String] virtual center database username
+        # @param vcdb_name [String] virtual center database name
+        # @param vc_sym_key [String] sym key from virtual center
+        # @return [Array] list of hash tables where each table is a user, nil on error
+        #
+        def get_vpx_vms(pg_password, vcdb_user, vcdb_name, _vc_sym_key)
+          return nil unless command_exists? psql_bin
+
+          output = []
+          vm_rows = cmd_exec("#{postgress_connect(pg_password, vcdb_user, vcdb_name)} -c 'SELECT vmid, name, configfilename, guest_state, is_template FROM vpxv_vms;' -P pager -A -t")
+          return nil if vm_rows.nil?
+
+          vm_rows = vm_rows.split("\n")
+          return nil unless vm_rows.first
+
+          vm_rows.each do |vm_row|
+            row_data = vm_row.split('|')
+            next if row_data.length < 5 # shoudld always be 5 based on query, but this will catch 'command not found' or other things like that
+
+            vm = {
+              'vmid' => row_data[0],
+              'name' => row_data[1],
+              'configfilename' => row_data[3],
+              'guest_state' => row_data[4],
+              'is_template' => row_data[5]
+            }
+            output.append(vm)
+          end
+          output
+        end
+
+        #
+        # Returns a hash table of the vcdb.properties file
+        # @param location [String] where the file is located. defaults to /etc/vmware-vpx/vcdb.properties
+        # @return [Hash] hash of the file contents, nil on error
+        #
+        def process_vcdb_properties_file(location = '/etc/vmware-vpx/vcdb.properties')
+          return nil unless file_exist?(location)
+
+          contents = read_file(location)
+          return nil if contents.nil?
+
+          output = {}
+          contents.split("\n").each do |line|
+            next unless line.include?('=') # attempt to do a little quality control
+
+            line = line.split('=')
+            key = line.shift.strip
+            value = line.join('=').strip
+            output[key] = value
+            next unless key == 'url'
+
+            # url is a compound object with database type, host, port, and name.
+            # we'll split that into its own as well to make them easy to reference
+            # example line -> 'jdbc:postgresql://localhost:5432/VCDB'
+            value = value.split('://')
+            output['db_engine'] = value[0].split(':')[1]
+            output['host'] = value[1].split(':')[0]
+            output['port'] = value[1].split(':')[1].split('/')[0]
+          end
+          # pull out the name from the url
+          unless output['url'].nil?
+            output['name'] = output['url'].split('/').last
```

Comment:
@h00die : it may have also form of `jdbc:postgresql://localhost:5432/VCDB?sslmode=disable` 
in that case additional - sort of - `split('?")[0]` required.